testsupport/spacebindingrequest.go

```diff
@@ -0,0 +1,50 @@
+package testsupport
+
+import (
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SpaceBindingRequestOption func(request *toolchainv1alpha1.SpaceBindingRequest)
+
+func WithSpecSpaceRole(spaceRole string) SpaceBindingRequestOption {
+	return func(s *toolchainv1alpha1.SpaceBindingRequest) {
+		s.Spec.SpaceRole = spaceRole
+	}
+}
+
+func WithSpecMasterUserRecord(mur string) SpaceBindingRequestOption {
+	return func(s *toolchainv1alpha1.SpaceBindingRequest) {
+		s.Spec.MasterUserRecord = mur
+	}
+}
+
+func WithNamespace(ns string) SpaceBindingRequestOption {
```

Comment:
right , that how I wanted to name it too but that function already exists for SpaceRequests in the same package: https://github.com/codeready-toolchain/toolchain-e2e/blob/82d14a8198f213081614ef91fe7579bc20eaa314/testsupport/spacerequest.go#L37-L41 

Maybe I can make this more generic and use it for both ?

Reply:
Alternatively we could group the files under `testsupport` folder into smaller packages , for example we can move:
-  `space.go` , `spacerequest.go` into a new folder `testsupport/space`
-`spacebinding.go` , `spacebindingrequest.go` into `testsupport/spacebinding`

but I suggest we do this in a separate PR.

Reply:
I like the `With` prefix because it's the same for all `Option`.
There's a pattern that works in case we want to keep everything in the same package, but it requires a bit of refactoring:

Something like:

```

type SpaceOption interface {
  ApplyToSpace(s *Space)
}

type SpaceRequestOption interface {
  ApplyToSpaceRequest(sr *SpaceRequest)
}

type WithNamespace struct {
  ns string
}

var _ SpaceOption = WithNamespace{}  // requires a `ApplyToSpace(s *Space)` method 

func (w WithNamespace) ApplyToSpace(s *Space) {...}

var _ SpaceRequestOption = WithNamespace{} // requires a `ApplyToSpaceRequest(s *SpaceRequest)` method 

func (w WithNamespace) ApplyToSpaceRequest(sr *SpaceRequest) {...}

```
hence, we can have the same `WithNamespace(...)` option to initialize Spaces and SpaceRequests resources.

Reply:
that being said, having a sub package for each type is fine and probably easier to do, given the current codebase!

Reply:
@xcoulon thanks a lot! indeed your pattern proposal is another option. I don't have a strong opinion, even if looking at it probably the package segregation would be "easier" ( just changing the import statements ), while this new pattern I'm not sure I saw it used anywhere else in our code ? 

My proposal is to decide which path we want to go and I'll address it in a follow up PR. WDYT ?

Reply:
I believe that splitting (or segregating) the code in `testsupport/space` + `testsupport/spacebinding` is probably the easiest thing to do for now, so let's go with that. Also, that's what we also did in [toolchain-common/pkg/test](https://github.com/codeready-toolchain/toolchain-common/tree/master/pkg/test) anyways (see all the subfolders for masteruserrecords, etc.)

Reply:
Ok, thanks ! I'm adding that to my backlog and will come up with a PR once I've merged the existing ones.